builtin/nomad/task.go

```diff
@@ -286,9 +288,90 @@ func (p *TaskLauncher) StartTask(
 func (p *TaskLauncher) WatchTask(
 	ctx context.Context,
 	log hclog.Logger,
+	ui terminal.UI,
 	ti *TaskInfo,
 ) (*component.TaskResult, error) {
-	return nil, status.Errorf(codes.Unimplemented, "WatchTask not implemented")
+	// We'll query for the allocation in the namespace of our task launcher
+	queryOpts := &api.QueryOptions{Namespace: p.config.Namespace}
+
+	// Accumulate our result on this
+	var result component.TaskResult
+
+	if client, err := getNomadClient(); err != nil {
+		return nil, err
+	} else {
```

Comment:
We might not need to nest if statements since it looks like we are returning on failure in many of them.

Reply:
Agreed - this should be refactored to remove the elses. It'll simplify this function by a lot and remove the extra nesting.